scalanet/src/io/iohk/scalanet/codec/ECDSASignatureCodec.scala

```diff
@@ -0,0 +1,36 @@
+package io.iohk.scalanet.codec
+
+import java.math.BigInteger
+
+import io.iohk.scalanet.crypto.ECDSASignature
+import scodec.Attempt.{Failure, Successful}
+import scodec.bits.{BitVector, ByteVector}
+import scodec.{Attempt, Codec, DecodeResult, SizeBound}
+import scodec.codecs.implicits.implicitByteVectorCodec
+
+object ECDSASignatureCodec extends Codec[ECDSASignature] {
+  override def decode(bits: BitVector): Attempt[DecodeResult[ECDSASignature]] = {
+    implicitByteVectorCodec.decode(bits) match{
+      case Failure(f) => Failure(f)
```

Comment:
Here you are failing on first failure, which is exact semantic of flatMap  for attempt. This means you can use for comprehensions and not pattern match on each step. example:
```
val a = Codec[Int]
val b = Codec[String]

val c = a.encode(5).require ++ b.encode("hello").require

val d = for {
  a1 <- a.decode(c)
  a2 <- b.decode(a1.remainder)
} yield (a1.value, a2.value)
```

which results in:
```
a: scodec.Codec[Int] = 32-bit signed integer
b: scodec.Codec[String] = string32(UTF-8)

c: scodec.bits.BitVector = BitVector(104 bits, 0x000000050000000568656c6c6f)

d: scodec.Attempt[(Int, String)] = Successful((5,hello))
```